### PR TITLE
DM-45887: show outer loop values

### PR DESCRIPTION
**Pull request by pkubanek**
- **Display outerLoopData - OuterLoopPageWidget**
- **DataFormater class**
- **MaxFormator, MaxMilliSeconds**
- **AxisValue**
- **AxisValue scale parameter**
- **Fix colors usage**
- **Documentation of new classes.**

### CODE REVIEW

File: python/lsst/ts/criopy/gui/formators.py [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+        Convert values to this unit. Default is None - no unit. If provided,
+        unit must be provided as well.
+    is_warn_func : `func`, optional
+        Function evaluated on each value. If true is returned, value is assumed
+        to be in warning range and will be color coded (displayed in warning
+        text). Default is None - no color coded warning value.
+    is_err_func : `func`, optional
+        Function evaluated on each value. If true is returned, value is assumed
+        to be in warning range and will be color coded (displayed in warning
+        text). Default is None - no color coded error value.
```
Comment (teweitsai) [inline python/lsst/ts/criopy/gui/formators.py:57]: The doc string of `is_err_func` has the `warning`. Maybe:

```
Function evaluated on each value. If true is returned, value is assumed
to be in error range and will be color coded (displayed in error
text). Default is None - no color coded error value.
```

File: python/lsst/ts/criopy/gui/formators.py [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+        Parameters
+        ----------
+        data : `BaseMsgType`
+            New data, in SALobj/ts_xml structure.
+
+        Returns
+        -------
+        text : `str`
+            String to display in the label."""
+        assert self._field is not None
```
Comment (teweitsai) [inline python/lsst/ts/criopy/gui/formators.py:126]: Returns
        -------
        text : `str`
            String to display in the label.
"""

File: python/lsst/ts/criopy/gui/formators.py [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+        if self.is_err_func is not None and self.is_err_func(value):
+            text = "<font color='{Colors.ERROR.name()}'>{text}</font>"
+        elif self.is_warn_func is not None and self.is_warn_func(value):
+            text = f"<font color='{Colors.WARNING.name()}'>{text}</font>"
+        return text
+
+    @Slot()
+    def reset_formator(self) -> None:
+        """Called to reset collected data."""
+        pass
```
Comment (teweitsai) [inline python/lsst/ts/criopy/gui/formators.py:140]: Is this required? If yes, please help to add the comment to explain.

**Review by teweitsai — APPROVED** [review]
Looks good.

File: python/lsst/ts/criopy/gui/formators.py (hunk 2 above)
Comment (pkubanek) [reply python/lsst/ts/criopy/gui/formators.py]: Changed according to Numpy doc guide to :
```
Returns
-----------
str
    String..
```

**Pull request merged**